suite/tests/api/drdecode_aarch64.c

```diff
@@ -82,6 +82,8 @@ test_mov_instr_addr(void)
     instrlist_encode(GD, ilist, generated_code, true);
     protect_mem(generated_code, gencode_max_size, ALLOW_EXEC | ALLOW_READ);
 
+    __builtin___clear_cache(generated_code, generated_code + gencode_max_size);
```

Comment:
`suite/tests/api/ibl-stress.c` also dynamically generates code. Perhaps we should add the same fix there as well.

Reply:
I had a look there, and as it's not immediately trying to execute the code it generates and we haven't seen an issue there in any of the runs we've run.

Reply:
Maybe add a comment in `drdecode_aarch64.c` about when we need to flush the icache.

> as it's not immediately trying to execute the code it generates

So can the bug still manifest in `ibl-stress` if the icache is big enough that it continued to have the stale instrs?